mlflow/store/model_registry/dbmodels/models.py

```diff
@@ -204,3 +210,71 @@ def __repr__(self):
     # entity mappers
     def to_mlflow_entity(self):
         return RegisteredModelAlias(self.alias, self.version)
+
+
+class EncryptedString(TypeDecorator):
+    """
+    A custom SQLAlchemy type that encrypts data before storing in the database
+    and decrypts it when retrieving.
+    """
+
+    impl = String(1000)
+    cache_ok = True
+
+    def __init__(self):
+        super().__init__()
+        # Get encryption key from environment variable or generate one
+        # In production, this should come from a secure key management service
+        encryption_key = (
+            os.environ.get("MLFLOW_WEBHOOK_SECRET_ENCRYPTION_KEY") or Fernet.generate_key()
+        )
+        self.cipher = Fernet(encryption_key)
+
+    def process_bind_param(self, value, dialect):
+        if value is not None:
+            return self.cipher.encrypt(value.encode()).decode()
+        return value
+
+    def process_result_value(self, value, dialect):
+        if value is not None:
+            return self.cipher.decrypt(value.encode()).decode()
+        return value
+
+
+class SqlWebhook(Base):
+    __tablename__ = "webhooks"
+
+    webhook_id = Column(String(256), nullable=False)
+    name = Column(String(256), nullable=False)
+    description = Column(String(1000), nullable=True)
+    url = Column(String(500), nullable=False)
+    events = Column(JSON, nullable=False)  # Array of WebhookEvent values
+    status = Column(String(20), nullable=False, default="ACTIVE")
+    secret = Column(EncryptedString(), nullable=True)  # Encrypted storage for HMAC secret
+    creation_timestamp = Column(BigInteger, default=get_current_time_millis)
+    last_updated_timestamp = Column(BigInteger, nullable=True, default=None)
+    deleted_timestamp = Column(BigInteger, nullable=True, default=None)  # For soft deletes
+
+    __table_args__ = (PrimaryKeyConstraint("webhook_id", name="webhook_pk"),)
+
+    def __repr__(self):
+        return (
+            f"<SqlWebhook ({self.webhook_id}, {self.name}, {self.url}, "
+            f"{self.status}, {self.creation_timestamp})>"
+        )
+
+    def to_mlflow_entity(self):
+        # Convert events from JSON array
+        events = [WebhookEvent(event) for event in self.events]
+
+        return Webhook(
+            webhook_id=self.webhook_id,
+            name=self.name,
+            url=self.url,
+            events=events,
+            creation_timestamp=self.creation_timestamp,
+            last_updated_timestamp=self.last_updated_timestamp,
+            description=self.description,
+            status=WebhookStatus(self.status),
+            secret=None,  # Don't expose secret in response
```

Comment:
q: If we don't include the secret from the response of the SQL model registry store, how will we use the secret for webhook URL invocation?

Reply:
I was going to update this when we need to use the secret, but let me update this in this PR.